corehq/apps/data_interfaces/interfaces.py

```diff
@@ -85,12 +87,22 @@ def rows(self):
     @property
     def report_context(self):
         context = super(CaseReassignmentInterface, self).report_context
-        active_users = self.get_all_users_by_domain(user_filter=tuple(HQUserType.all()), simplified=True)
+        if not self.request.can_access_all_locations:
+            accessible_location_ids = (SQLLocation.active_objects.accessible_location_ids(
+                self.request.domain,
+                self.request.couch_user)
+            )
+            accessible_user_ids = user_ids_at_locations(accessible_location_ids)
+            active_users = self.get_all_users_by_domain(user_filter=tuple(HQUserType.all()), simplified=True,
+                                                        user_ids=tuple(accessible_user_ids))
```

Comment:
Thats right. Thank You @esoergel.
Using the first suggestion. I think `get_all_users_by_domain` just handles the case where a user is not found and gives a temp user but i don't think that would be an issue here.